### PR TITLE
Only use OTP based auth, do not expect password

### PR DESCRIPTION
**Pull request by jdk2588**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -92,15 +92,14 @@ func (h configHandler) Bind(bindDN, bindSimplePw string, conn net.Conn) (resultC
 	}
 
 	if len(user.OTPSecret) > 0 && !validotp {
-		if len(bindSimplePw) > 6 {
-			otp := bindSimplePw[len(bindSimplePw)-6:]
-			bindSimplePw = bindSimplePw[:len(bindSimplePw)-6]
-
-			validotp = totp.Validate(otp, user.OTPSecret)
+		if len(bindSimplePw) == 6 {
```
Comment (meson10) [inline config.go]: Either add a switch per server or make it a part of the user json.

Comment (meson10) [reply config.go]: example `--otp-length=6`

File: config.go [hunk 2]
```diff
@@ -92,15 +92,14 @@ func (h configHandler) Bind(bindDN, bindSimplePw string, conn net.Conn) (resultC
 	}
 
 	if len(user.OTPSecret) > 0 && !validotp {
-		if len(bindSimplePw) > 6 {
-			otp := bindSimplePw[len(bindSimplePw)-6:]
-			bindSimplePw = bindSimplePw[:len(bindSimplePw)-6]
-
-			validotp = totp.Validate(otp, user.OTPSecret)
+		if len(bindSimplePw) == h.cfg.OTPlength {
```
Comment (meson10) [inline config.go]: Your configuration has a default int. and the length will be 0.
So technically a user could press blank Enter and get in, If a  configuration didn't supply the value.

Comment (jdk2588) [reply config.go]: it will not because the 2fa library will validate for the seed and it will not return anything

File: config.go [hunk 3]
```diff
@@ -92,15 +92,14 @@ func (h configHandler) Bind(bindDN, bindSimplePw string, conn net.Conn) (resultC
...
+			validotp = totp.Validate(bindSimplePw, user.OTPSecret)
 		}
 	}
 
-	if !validotp {
+	if validotp {
+		return ldap.LDAPResultSuccess, nil
+	} else {
```
Comment (meson10) [inline config.go]: The curly braces are needless. It will return anyway. Bring back one level of Indent.

**Pull request merged**